test/built-ins/BigInt/asIntN/bits-toindex.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2017 Josh Wolfe. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: pending
+description: BigInt.asIntN type coercion for bits parameter
+info: >
+  BigInt.asIntN ( bits, bigint )
+
+  1. Let bits be ? ToIndex(bits).
+features: [Symbol]
+---*/
+
+function MyError() {}
+
+assert.sameValue(BigInt.asIntN(undefined, 1n), 0n);
+assert.sameValue(BigInt.asIntN(null, 1n), 0n);
+assert.sameValue(BigInt.asIntN(true, 1n), -1n);
+assert.sameValue(BigInt.asIntN(false, 1n), 0n);
+assert.sameValue(BigInt.asIntN("foo", 1n), 0n);
+assert.sameValue(BigInt.asIntN("3", 10n), 2n);
+assert.throws(TypeError, () => BigInt.asIntN(Symbol(0), 0n));
+assert.throws(TypeError, () => BigInt.asIntN(1n, 0n));
+assert.sameValue(BigInt.asIntN(Object(3), 10n), 2n);
+assert.sameValue(BigInt.asIntN({valueOf:()=>3}, 10n), 2n);
```

Comment:
I don't think checking `Object(3)` is necessary as we already have this custom object here. Removing the previous line (with the Object call) should be fine, otherwise, it could be explicit using `new Number`.

Reply:
I just noticed that `? GetMethod(input, @@toPrimitive)` can throw a TypeError if the `[Symbol.toPrimitive]` is not callable. I should probably add a test for that too.

I'm getting nervous about writing such thorough tests under the name "BigInt" for an aspect of the spec that has nothing to do with the BigInt proposal. This kind of thing makes #1197 appealing.

Reply:
in a user level, this might not seem relevant, but the purpose of this project is to confirm the observable steps in the spec, in the order they are defined. This is crucial to mitigate inconsistency from browsers and even detect issues like those in the web reality matters.

In the moment we say it's relevant for the spec to say a TypeError should be thrown if `@@toPrimitive` is not callable, a test for that is important, and that's a part of the BigInt api being covered here.

Although, it's not necessary to write a complete coverage in a single Pull Request, it can be fixed through follow up Pull Requests and I'm happy to volunteer writing those.